armi/bookkeeping/db/database3.py

```diff
@@ -1264,11 +1264,18 @@ def _readParams(h5group, compTypeName, comps):
                 pDef = pDefs[paramName]
             except KeyError:
                 if re.match(r"^n[A-Z][a-z]?\d*", paramName):
-                    # This is a temporary viz param made by _addHomogenizedNumberDensities
-                    # ignore it safely
+                    # This is a temporary viz param (number density) made by
+                    # _addHomogenizedNumberDensities ignore it safely
                     continue
                 else:
-                    raise
+                    # If a parameter exists in the database but not in the application
+                    # reading it, we can technically keep going. Since this may lead to
+                    # potential correctness issues, raise a warning
+                    runLog.warning(
+                        "Found `{}` parameter `{}` in the database, which is not defined. "
+                        "Ignoring it.".format(compTypeName, paramName)
+                    )
+                    continue
```

Comment:
This seems like we should add a unit test in `test_database3.py` to capture that this no longer raises an error for missing parameters.

Reply:
>What is this about "correctness issues"? If the program doesn't need the data, the data won't affect how "correct" the program is, right?

Correctness issues as in loading a data serialization file from at best a different version of the same program (where mechanisms to handle added/removed/renamed parameters are not explicitly specified), or perhaps a different application entirely, in which there's no guaranteed correspondence between the parameters that were written to the database and those that are defined in the application reading it. The last seems contrived, but most requests that I've gotten to relax this check have been to allow reading databases that were produced from completely different apps.

Reply:
Sidenote, this was kind of the inspiration for #191